### ISSUE TITLE
Make REST error behavior customizable

### ISSUE THREAD

**Issue by amanessinger** [amanessinger/azzyzt_jee_tools#26]
Currently we catch errors that would lead to (correct) 500 and deliver a 200 OK with an error response. This is due to the way we use REST along with a home-grown Flex network interface to cope with restrictions in our portal. Nobody else will need that, and worse, the response is hard-coded to be XML. JSON requests should always give JSON responses.

**Comment by amanessinger** [comment]
Even worse: while using the generated Apache CXF REST client for unit tests, I found that this behavior is simply unacceptable for Java REST clients. CXF fails with an exception, because there is a response, but not a response that could be parsed into the type indicated by the interface.

Solution: we require a header "X-Portal-Cred??" (with "??" being a number) to be set to "REST_200_ON_ERROR". This is what happens in our environment, when we set "attributes" in the portal. If the header is not present, the server acts normally. 

Thus we require a change from all our users, but the change is not intrusive. Flex clients would access the application via another portal URL than "normal" clients. The portal would map both URLs to the same actual service. The default would be sane behavior.
